pkgs/by-name/hy/hydra-check/package.nix

```
@@ -23,6 +24,13 @@ rustPlatform.buildRustPackage rec {
   useFetchCargoVendor = true;
   cargoHash = "sha256-G9M+1OWp2jlDeSDFagH/YOCdxGQbcru1KFyKEUcMe7g=";
 
+  patches =
+    lib.optional
+      (lib.hasPrefix "1.88" buildPackages.rustc.version && stdenv.buildPlatform.system == "x86_64-darwin")
```

Thanks for downstreaming this patch :). I don't like this approach though - why not apply the patch unconditionally? I'll break it down per condition:

- `stdenv.buildPlatform.system ==`: It'd be easier for everyone to know whether the test applies or not, if the patch would be applied without this condition.
- `lib.hasPrefix "1.88" buildPackages.rustc.version`: Do we have an idea for when rustc 1.89 will reach Nixpkgs? If the patch will be applied never the less, would it be bad for rustc 1.89? I think we can trust upstream to update their Cargo.toml file when rustc 1.89 will reach nixpkgs `master`?

---

Since this is specifically an **x86_64-darwin** compiler / linker bug, I think it would be best to keep the platform condition and drop the rust version condition. In particular, note that rustc 1.89 will be released around Aug 7 and it contains the fix for this compiler / linker bug, but:
- I am not sure when it will enter nixpkgs
- It has been confirmed that the bug is fixed in rustc nightly but no one has really tested it against 1.89 stable (well of course, it hasn't been released yet)
- This PR basically compiles a key dependency (hyper) without optimizations. This will hurt performance, but probably not by that much, so we can keep the patch even for rustc 1.89

One thing I noticed after carefully reading #427072 and the links there-in is that the patch should be conditioned on `hostPlatform` = `x86_64-darwin` instead of the build platform. So how about this:
```suggestion
  patches =
    lib.optional
      (stdenv.hostPlatform.system == "x86_64-darwin")
```

---

Hmm OK since this is could be a performance penalty, maybe it _is_ a good reason to apply this patch only for (hostPlatform) `x86_64-darwin`. I have a question: Is it possible in the Rust ecosystem to write in a `Cargo.toml` file that only certain host platforms should compile dependencies differently?